packages/core/src/values/Primitive.ts

```diff
@@ -231,15 +235,17 @@ export function encodeFloatWithScale(
 }
 
 /** Parses a bit mask into a numeric array */
-export function parseBitMask(mask: Buffer, startValue: number = 1): number[] {
-	const numBits = mask.length * 8;
-
+export function parseBitMask(
+	mask: Buffer,
+	startValue: number = 1,
+	numBits: number = mask.length * 8,
+): number[] {
 	const ret: number[] = [];
-	for (let index = 1; index <= numBits; index++) {
-		const byteNum = (index - 1) >>> 3; // id / 8
-		const bitNum = (index - 1) % 8;
+	for (let index = 0; index < numBits; index++) {
```

Comment:
I personally found this confusing... we do 1 based indexing in the `for` statement, only to subtract off `1` everywhere below. I just removed the `1` everywhere.